packages/react/src/contents/hooks/useSeoFiles.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  fetchSEOFiles,
+  getSEOFilesError,
+  getSEOFilesResult,
+  isSEOFilesFetched,
+  isSEOFilesLoading,
+  type StoreState,
+} from '@farfetch/blackout-redux';
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../helpers/useAction.js';
+import type { UseSeoFilesOptions } from './types/useSeoFiles.types.js';
+
+/**
+ * Hook to return seo metadata to populate document head.
+ *
+ * @param data - object that receives SEO query, icon links, additional links, metas and options
+ *
+ * @returns - Returns actions and SEO metadata state for a given path.
+ */
+const useSeoFiles = ({ query, enableAutoFetch = true }: UseSeoFilesOptions) => {
+  const error = useSelector((state: StoreState) =>
+    getSEOFilesError(state, query),
+  );
+
+  const isLoading = useSelector((state: StoreState) =>
+    isSEOFilesLoading(state, query),
+  );
+
+  const isFetched = useSelector((state: StoreState) =>
+    isSEOFilesFetched(state, query),
+  );
+
+  const data = useSelector((state: StoreState) =>
+    getSEOFilesResult(state, query),
+  );
+
+  const fetchSEOFilesAction = useAction(fetchSEOFiles);
+
+  const fetch = useCallback(
+    () => fetchSEOFilesAction(query),
+    [fetchSEOFilesAction, query],
+  );
+
+  useEffect(() => {
+    if (enableAutoFetch && !isLoading && !error && !isFetched) {
```

Comment:
No need to check for error as it is implicitly guaranteed by the `isFetched` value. If there is an error, `isFetched` will be true.

```suggestion
    if (enableAutoFetch && !isLoading && !isFetched) {
```

Reply:
Resolved! ✅